src/main/java/com/ibm/eventstreams/connect/mqsource/builders/CustomizableJsonConverter.java

```diff
@@ -0,0 +1,883 @@
+        } catch (ClassCastException e) {
+            String schemaTypeStr = (schema != null) ? schema.type().toString() : "unknown schema";
+            throw new DataException("Invalid type for " + schemaTypeStr + ": " + value.getClass());
+        }
+    }
+
+
+    private static Object convertToConnect(Schema schema, JsonNode jsonValue) {
+        final Schema.Type schemaType;
+        if (schema != null) {
+            schemaType = schema.type();
+            if (jsonValue == null || jsonValue.isNull()) {
+                if (schema.defaultValue() != null) {
+                    return schema.defaultValue(); // any logical type conversions should already have been applied
+                }
+                if (schema.isOptional()) {
+                    return null;
+                }
+                throw new DataException("Invalid null value for required " + schemaType + " field");
+            }
+        } else {
+            switch (jsonValue.getNodeType()) {
+                case NULL:
+                    // Special case. With no schema
+                    return null;
+                case BOOLEAN:
+                    schemaType = Schema.Type.BOOLEAN;
+                    break;
+                case NUMBER:
+                    if (jsonValue.isIntegralNumber()) {
+                        schemaType = Schema.Type.INT64;
+                    } else {
+                        schemaType = Schema.Type.FLOAT64;
+                    }
+                    break;
+                case ARRAY:
+                    schemaType = Schema.Type.ARRAY;
+                    break;
+                case OBJECT:
+                    schemaType = Schema.Type.MAP;
+                    break;
+                case STRING:
+                    schemaType = Schema.Type.STRING;
+                    break;
+
+                case BINARY:
+                case MISSING:
+                case POJO:
+                default:
+                    schemaType = null;
+                    break;
+            }
+        }
+
+        final JsonToConnectTypeConverter typeConverter = TO_CONNECT_CONVERTERS.get(schemaType);
+        if (typeConverter == null) {
+            throw new DataException("Unknown schema type: " + String.valueOf(schemaType));
+        }
+
+        Object converted = typeConverter.convert(schema, jsonValue);
+        if (schema != null && schema.name() != null) {
+            LogicalTypeConverter logicalConverter = TO_CONNECT_LOGICAL_CONVERTERS.get(schema.name());
+            if (logicalConverter != null) {
+                converted = logicalConverter.convert(schema, converted);
+            }
+        }
+        return converted;
+    }
+
+    private interface JsonToConnectTypeConverter {
+
+        Object convert(Schema schema, JsonNode value);
+
+    }
+
+    private interface LogicalTypeConverter {
+
+        Object convert(Schema schema, Object value);
+
+    }
+
+    static class JsonSchema {
+
```

Comment:
added this as a static inner-class as normally it's package-private class for original JsonConverter